### PR TITLE
[FIX] Validate userstore update request inputs

### PR DESCRIPTION
**Pull request by somindatommy**
Resolves https://github.com/wso2/product-is/issues/9558

### Approach
- **Commit1:** refactoring the error codes
- **Commit2:** 
**Resolves:** https://github.com/wso2/product-is/issues/9558
**Approach:** Validating domain id and domain name

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java [hunk 1]
```diff
@@ -872,4 +875,36 @@ private APIError handleIdentityUserStoreException(IdentityUserStoreMgtException
...
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_REQUEST_BODY_NOT_FOUND.getMessage());
+        }
+        if (StringUtils.isBlank(userStoreReq.getName())) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getCode(),
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getMessage());
+        }
+        if (!userStoreReq.getName().equals(base64URLDecodeId(domainID))) {
+            throw new IdentityUserStoreClientException(
+                    UserStoreConstants.ErrorMessage.ERROR_CODE_EMPTY_DOMAIN_NAME.getCode(),
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java:906]: is this error code correct?

Comment (somindatommy) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.v1/src/main/java/org/wso2/carbon/identity/api/server/userstore/v1/core/ServerUserStoreService.java]: Fixed 0769df00c8e498d9efdbb1f8827130f28567c8de

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java [hunk 2]
```diff
@@ -75,22 +75,30 @@
...
-        ERROR_CODE_ERROR_RETRIEVING_PRIMARY_USERSTORE("65011",
-                "Unable to get the primary user store.",
-                "Server Encountered an error while retrieving the primary user store.",
-                Response.Status.INTERNAL_SERVER_ERROR);
+        ERROR_CODE_EMPTY_DOMAIN_ID("60007", "Userstore ID is not specified", "Userstore " +
+                "ID is either NULL or empty."),
+        ERROR_CODE_EMPTY_DOMAIN_NAME("60008", "Userstore domain name is not specified",
+                "Userstore domain name is either NULL or empty."),
+        ERROR_CODE_DOMAIN_ID_DOES_NOT_MATCH_WITH_NAME("60009", "Invalid userstore domain name or " +
+                "domainID", "Userstore domain name does not match with the userstore domain ID"),
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConstants.java:99]:
```suggestion
                "domain ID", "Userstore domain name does not match with the userstore domain ID"),
```

**Review by pulasthi7 — APPROVED** [review]

**Comment by somindatommy** [comment]
PR builder JOB
https://jenkins-support-all.wso2.com/jenkins/job/DEV_PR_BUILDER/914/

**Pull request merged**